### PR TITLE
feat(iOS, Tabs): integration with ScrollViewMarker

### PR DESCRIPTION
**Pull request by kkafar**
## Description

Integrates the `ScrollViewMarker` component with the Tabs special-effects system on iOS.

Until now, Tabs relied purely on a descendant-chain heuristic (`RNSScrollViewFinder`) to
locate the scroll view backing a tab screen. This PR lets a `ScrollViewMarker` explicitly
register its wrapped scroll view with the enclosing tab screen, so the *content scroll view*
is known precisely instead of being guessed. The heuristic is kept as a fallback when no
marker is present.

This integration is currently gated behind the `RNS_GAMMA_ENABLED` compilation flag and will
remain so until we are able to remove that flag.

Closes https://github.com/software-mansion/react-native-screens-labs/issues/1536
Closes https://github.com/software-mansion/react-native-screens-labs/issues/1537

## Changes

- `RNSTabsScreenComponentView` now conforms to `RNSScrollViewSeeking`. When a descendant
  marker registers its scroll view, the screen forwards it to its view controller via
  `setContentScrollView:forEdge:NSDirectionalRectEdgeAll`. On a tab change the
  `RNSTabBarController` queries the child view controller for its content scroll view, so the
  registered scroll view flows into the special-effects system automatically.
- `RNSTabsScreenViewController` gains `resolveContentScrollView`, which prefers the explicitly
  registered content scroll view (querying the bottom, then top edge) and falls back to the
  `RNSScrollViewFinder` heuristic. The repeated-tab-selection scroll-to-top effect now uses it.
- `RNSScrollViewMarkerComponentView`:
  - walks the React superview chain (`reactSuperview`) when looking for a seeking ancestor,
    because at transaction-end time the `RNSTabsScreenComponentView` may not yet be attached to
    its parent (tab not selected, or registration runs too early);
  - resets its registration state when it leaves the window, so it re-registers on
    re-attachment (e.g. switching back to a tab).
- Cleanup in `RNSTabsHostComponentView`.
- Added an integration test scenario exercising a marker inside Tabs.

### Notes / limitations

- Only the case where the `ScrollViewMarker` sits directly under Tabs (no Stack v4/v5 in
  between) is handled here. Those compositions will be integrated in follow-up PRs.
- With multiple `ScrollViewMarker`s under the same seeking ancestor there is currently a
  potential race condition in which marker wins registration.

### Edge effects

> [!important]
> Seemingly the edge effect is applied correctly no matter the integration with the scrollview marker. Likely `UITabBarController` uses some different logic to `UIViewController.contentScrollView` to detect scrollview, because even when I have had returned `nil` from the method, the edge effect had still been applied. Nevertheless, I decided to include this test case here, just to make sure it works. Also this allows us to test that different ScrollViewMarkers from different tab update the edge effect correctly on tab change.

| before | after | 
| -- | -- | 
| <video src="https://github.com/user-attachments/assets/3afacdec-9dd5-4d24-8272-d2e5e8dd8009" alt="scroll-edge-effects-before" /> | <video src="https://github.com/user-attachments/assets/d0b0bf17-5be2-45ee-8615-e1e7b48f0214" alt="scroll-edge-effects-after" /> | 

### Special effects 

| before | after | 
| -- | -- | 
| <video src="https://github.com/user-attachments/assets/c5a1fc25-99cd-4c10-a915-bdad7c9cbe31" alt="tabs-special-effects-before" /> | <video src="https://github.com/user-attachments/assets/c5813243-eab4-4698-81bf-671de6c64edf" alt="tabs-special-effect-after" /> | 

## Test plan

Tested with the new integration scenario:

- `apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-edge-effects/index.tsx`
- `apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-edge-effects/scenario-description.ts`
- registered in `apps/src/tests/component-integration-tests/index.tsx` as `ScrollViewMarker`

Steps (iOS, `RNS_GAMMA_ENABLED=1`):

1. Run `FabricExample` and open the component-integration tests.
2. Open `ScrollViewMarker integration cases` → `Integration: SVM in Tabs - edge effects`.
3. Scroll the list in each tab and confirm the scroll edge effect is applied per tab
   (`Home` → `hard`, `Second` → `soft`).
4. Tap the currently selected tab again and confirm the list scrolls to top (repeated-tab
   selection effect resolves the marker-registered scroll view).

## Checklist

- [x] Included code example that can be used to test this change.
- [x] For visual changes, included screenshots / GIFs / recordings documenting the change.
- [x] For API changes, updated relevant public types.
- [x] Ensured that CI passes

### CODE REVIEW

File: ios/tabs/screen/RNSTabsScreenViewController.mm [hunk 1]
```diff
@@ -97,14 +97,23 @@ - (bool)tabScreenSelectedRepeatedly
...
+- (nullable UIScrollView *)resolveContentScrollView
+{
+  if (auto sv = [self contentScrollViewForEdge:NSDirectionalRectEdgeBottom]; sv != nil) {
+    return sv;
+  }
+  if (auto sv = [self contentScrollViewForEdge:NSDirectionalRectEdgeTop]; sv != nil) {
+    return sv;
+  }
+  return [RNSScrollViewFinder findScrollViewInFirstDescendantChainFrom:[self tabScreenComponentView]];
+}
```
[the remark this answers is not in the document]
Comment (kkafar) [reply ios/tabs/screen/RNSTabsScreenViewController.mm:115]: These are regular UIKit methods.

File: ios/tabs/screen/RNSTabsScreenComponentView.mm [hunk 2]
```diff
@@ -100,10 +115,25 @@ - (void)invalidateImpl
...
+#pragma mark RNSScrollViewSeeking
+
+#if RNS_GAMMA_ENABLED
+- (void)registerDescendantScrollView:(UIScrollView *)scrollView fromMarker:(RNSScrollViewMarkerComponentView *)marker
+{
+  _markerContentScrollView = scrollView;
+  _scrollViewMarker = marker;
+  [_controller setContentScrollView:scrollView forEdge:NSDirectionalRectEdgeAll];
+}
+#endif // RNS_GAMMA_ENABLED
```
[the remark this answers is not in the document]
Comment (kkafar) [reply ios/tabs/screen/RNSTabsScreenComponentView.mm:124]: This is a regular UIKit method.

File: ios/gamma/scroll-view-marker/RNSScrollViewMarkerComponentView.mm [hunk 3]
```diff
@@ -78,7 +79,11 @@ - (nullable UIScrollView *)findScrollView
     if ([superview respondsToSelector:@selector(registerDescendantScrollView:fromMarker:)]) {
       return static_cast<id<RNSScrollViewSeeking>>(superview);
     }
-    superview = superview.superview;
+    if ([superview respondsToSelector:@selector(reactSuperview)]) {
+      superview = [superview reactSuperview];
+    } else {
+      superview = superview.superview;
+    }
```
[the remark this answers is not in the document]
Comment (kkafar) [reply ios/gamma/scroll-view-marker/RNSScrollViewMarkerComponentView.mm:86]: That is right. 

Indeed we have a bug in `RNSTabsHostComponentView` -> it never populates the `_reactSuperview` field nor it uses it nor anything. Here's [ticket for it](https://github.com/software-mansion/react-native-screens-labs/issues/1539).

As the other concern goes -> that's ok.

[the remark this answers is not in the document]
Comment (kkafar) [reply ios/gamma/scroll-view-marker/RNSScrollViewMarkerComponentView.mm:86]: No, it works. Host populates that field for its child.

[the remark this answers is not in the document]
Comment (kkafar) [reply ios/gamma/scroll-view-marker/RNSScrollViewMarkerComponentView.mm:86]: That works fine. See the issue for explanation.

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario-description.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import type { ScenarioDescription } from '@apps/tests/shared/helpers';
+
+export const scenarioDescription: ScenarioDescription = {
+  name: 'Integration: SVM in Tabs - edge effects',
```
Comment (kligarski) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario-description.ts]: nit: do we need the "Integration:" in the name if it's already in component integration tests directory? Applies to both test screens.

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+iOS 26.5, Android SDK 36
+
+## E2E test
+
+TBD
+
+## Prerequisites
+
+iOS: simulator with iOS 26+ is enough,
+Android: emulator
```
Comment (kligarski) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md]:
```suggestion
```
I don't think there are any scroll edge effects on Android.

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+## Details
+
+**Description:**
+This test verifies interaction between `ScrollViewMarker` and `Tabs` in scope of scroll-edge-effects.
+It allows to test both whether the scroll-edge-effect is correctly applied AND whether it is correctly 
+updated between different tabs. 
+
+**OS test creation version:** 
+iOS 26.5, Android SDK 36
```
Comment (kligarski) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md]:
```suggestion
iOS 26.5
```

File: ios/tabs/screen/RNSTabsScreenViewController.mm [hunk 7]
```diff
@@ -97,14 +97,23 @@ - (bool)tabScreenSelectedRepeatedly
...
 }
 
+- (nullable UIScrollView *)resolveContentScrollView
+{
+  if (auto sv = [self contentScrollViewForEdge:NSDirectionalRectEdgeBottom]; sv != nil) {
+    return sv;
+  }
+  if (auto sv = [self contentScrollViewForEdge:NSDirectionalRectEdgeTop]; sv != nil) {
+    return sv;
+  }
```
Comment (kligarski) [inline ios/tabs/screen/RNSTabsScreenViewController.mm:113]: Is there a reason why we check bottom and then top edge in this order or is it arbitrarily chosen? It would be nice to have a comment here.

File: ios/gamma/scroll-view-marker/RNSScrollViewMarkerComponentView.mm [hunk 8]
```diff
@@ -150,6 +155,14 @@ - (void)willMoveToWindow:(UIWindow *)newWindow
   [self maybeRegisterWithSeekingAncestor];
 }
 
+- (void)didMoveToWindow
+{
+  [super didMoveToWindow];
+  if (self.window == nil) {
+    _hasAttemptedRegistration = NO;
```
Comment (kligarski) [inline ios/gamma/scroll-view-marker/RNSScrollViewMarkerComponentView.mm:162]: The marker immediately re-registers the scroll view after any mounting transaction:

https://github.com/user-attachments/assets/c7d946a9-c7d8-4085-96b0-0e6d315bc05b

Should we gate registration with `window != nil` check maybe?

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+      </ScrollViewMarker>
+    </View>
+  );
+}
+
+export function TabsNavigation() {
+  return <TabsContainer routeConfigs={TABS_ROUTE_CONFIGS} />;
+}
+
+function StackTabScreen() {
```
Comment (t0maboro) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx:97]: aren't we missing `export` or is this intentional?

**Comment by t0maboro** [comment]
I'm seeing some inconsistency: after the 1st tab switch, scrollToTop isn't working, after subsequent switches it started working

<video src="https://github.com/user-attachments/assets/74de2d13-cb65-4b8f-8c83-fe8ff95090a3" />

**Review by t0maboro — APPROVED** [review]

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+## Details
+
+**Description:**
+This test verifies interaction between `ScrollViewMarker`, `Stack` and `Tabs` components. 
+The primary goal is to verify that the tabs special effects (scroll-to-top, pop-to-top) do work in 
+nested container scenario.
+
+**OS test creation version:** 
+iOS 26.5, Android SDK 36
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md]: To unify it with other test scenarios:

```suggestion
iOS: 26.5, Android: API Level 36.
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/index.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+import TestSvmTabsScrollEdgeEffects from './test-svm-tabs-scroll-edge-effects';
+import TestStackSvmTabsSpecialEffects from './test-stack-svm-tabs-special-effects';
+
+const scenarios = {
+  TestSvmTabsScrollEdgeEffects,
+  TestStackSvmTabsSpecialEffects,
+};
+
+const TestSvmScenarioGroup: ScenarioGroup<keyof typeof scenarios> = {
+  name: 'ScrollViewMarker integration cases',
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/index.ts]: We should unify title with already existing CIT scenarios:

```suggestion
  name: 'ScrollViewMarker Integration Tests',
```

File: apps/src/tests/component-integration-tests/index.tsx [hunk 12]
```diff
@@ -11,13 +11,15 @@ import OrientationScenarioGroup from './orientation';
 import ScrollViewScenarioGroup from './scroll-view';
 import FormSheetScenarioGroup from './form-sheet';
 import TabsInStackV5ScenarioGroup from './tabs-stack-v5';
+import TestSvmScenarioGroup from './scroll-view-marker';
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/index.tsx]: For Scenario Groups I would keep the name without 'Test', making it here SvmScenarioGroup or ScrollViewMarkerScenarioGroup

```suggestion
import SvmScenarioGroup from './scroll-view-marker';
```

File: apps/src/tests/component-integration-tests/index.tsx [hunk 13]
```diff
@@ -11,13 +11,15 @@ import OrientationScenarioGroup from './orientation';
...
 import ScenarioSelectionScreen from '@apps/tests/shared/ScenarioScreen';
 
 export const COMPONENT_SCENARIOS = {
   Orientation: OrientationScenarioGroup,
   ScrollView: ScrollViewScenarioGroup,
   FormSheet: FormSheetScenarioGroup,
   TabsInStackV5: TabsInStackV5ScenarioGroup,
+  ScrollViewMarker: TestSvmScenarioGroup,
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/index.tsx]: Depends which name will be chosen above it has to be changed also here

```suggestion
  ScrollViewMarker: SvmScenarioGroup,
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx (hunk 9 above)
Comment (LKuchno) [reply apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx:97]: we are coming back to state where we will export only App() (now it should be renamed) and default export createScenario()
https://github.com/software-mansion/react-native-screens/pull/4201

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+import { Colors } from '@apps/shared/styling';
+import { Rectangle } from '@apps/shared/Rectangle';
+import { ScrollViewMarker } from 'react-native-screens/experimental';
+import { type ScrollEdgeEffect } from 'react-native-screens';
+import {
+  StackContainer,
+  type StackRouteConfig,
+} from '@apps/shared/gamma/containers/stack';
+
+export function App() {
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx]: To align with changes from PR#4201 rename App to test screen name:

```suggestion
export function TestStackSvmTabsSpecialEffects() {
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+      })}
+    </>
+  );
+}
+
+function HeuristicBreakingView() {
+  return <View collapsable={false} style={{ width: '100%', height: 0 }}></View>;
+}
+
+export default createScenario(App, scenarioDescription);
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx]:
```suggestion
export default createScenario(TestStackSvmTabsSpecialEffects, scenarioDescription);
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+
+export function TabsNavigation() {
+  return <TabsContainer routeConfigs={TABS_ROUTE_CONFIGS} />;
+}
+
+function StackTabScreen() {
+  return <StackContainer routeConfigs={STACK_ROUTE_CONFIGS} />;
+}
+
+export function ScrollViewContents(props: { elementCount?: number }) {
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx]: As mentioned above:

```suggestion
function ScrollViewContents(props: { elementCount?: number }) {
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+          contentInsetAdjustmentBehavior="automatic"
+          style={styles.fillParent}>
+          <ScrollViewContents />
+        </ScrollView>
+      </ScrollViewMarker>
+    </View>
+  );
+}
+
+export function TabsNavigation() {
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx]: As mentioned above:

```suggestion
function TabsNavigation() {
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx [hunk 18]
```diff
@@ -0,0 +1,133 @@
...
+    options: {},
+  },
+  {
+    name: 'Second',
+    Component: StackContents,
+    options: {},
+  },
+];
+
+export function TabContents() {
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx]: I think we can remove all exports here (except exporting TestStackSvmTabsSpecialEffects and default export of createScenario).

```suggestion
function TabContents() {
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx [hunk 19]
```diff
@@ -0,0 +1,133 @@
...
+          contentInsetAdjustmentBehavior="automatic"
+          style={styles.fillParent}>
+          <ScrollViewContents />
+        </ScrollView>
+      </ScrollViewMarker>
+    </View>
+  );
+}
+
+export function StackContents() {
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/index.tsx]: As mentioned above:

```suggestion
function StackContents() {
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario-description.ts [hunk 20]
```diff
@@ -0,0 +1,12 @@
+import type { ScenarioDescription } from '@apps/tests/shared/helpers';
+
+export const scenarioDescription: ScenarioDescription = {
+  name: 'Integration: SVM in Stack & Tabs - tabs special effects',
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario-description.ts]: There is no need to put 'Integration' into name as the test is under CIT directory.

```suggestion
  name: 'SVM in Stack & Tabs - tabs special effects',
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario-description.ts [hunk 21]
```diff
@@ -0,0 +1,12 @@
...
+  key: 'test-stack-svm-tabs-special-effects',
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario-description.ts:5]: Im not sure if the name shouldn't be like this:

```suggestion
  key: 'test-svm-in-stack-tabs-special-effects',
```

It's what I assumed from scenario name. If I'm right, please rename also test directory.

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md [hunk 22]
```diff
@@ -0,0 +1,43 @@
+# Test Scenario: Integration: SVM in Stack & Tabs - tabs special effects
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md]: As in scenario-description file I would delete 'Integration:'
```suggestion
# Test Scenario: SVM in Stack & Tabs - tabs special effects
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md [hunk 23]
```diff
@@ -0,0 +1,43 @@
...
+iOS 26.5, Android SDK 36
+
+## E2E test
+
+TBD
+
+## Prerequisites
+
+iOS: simulator with iOS 15+ is enough,
+Android: emulator
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md]: Just formatting 

```suggestion
- iOS: simulator with iOS 15+ is enough,
- Android: emulator
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+
+## Note
+
+Android part of the code is unimplemented yet, therefore it is expected not to work.
+
+iOS implementation doesn't currently support nested container interaction yet. 
+This test needs to be updated after such interaction is supported.
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md:27]: Formatting: 

```suggestion
- Android part of the code is unimplemented yet, therefore it is expected not to work.
- iOS implementation doesn't currently support nested container interaction yet. 
- This test needs to be updated after such interaction is supported.
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+## Steps
+
+1. Open the test.
+
+- [ ] There should be two tabs: `Home` and `Stack`.
+- [ ] `Home` tab should be selected.
+
+2. Scrolldown a bit.
+
+    Doesn't really matter how much you scroll - the distance should be "noticeable".
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md]:
```suggestion
2. Scrolldown a bit - the distance should be "noticeable". 
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/index.tsx [hunk 26]
```diff
@@ -0,0 +1,96 @@
...
+  type TabRouteConfig,
+  TabsContainer,
+  useTabsNavigationContext,
+} from '@apps/shared/gamma/containers/tabs';
+import { Colors } from '@apps/shared/styling';
+import { Rectangle } from '@apps/shared/Rectangle';
+import { ScrollViewMarker } from 'react-native-screens/experimental';
+import { type ScrollEdgeEffect } from 'react-native-screens';
+
+export function App() {
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/index.tsx]: To align with changes from PR#4201 rename App to test screen name:

```suggestion
export function TestSvmTabsScrollEdgeEffects() {
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/index.tsx [hunk 27]
```diff
@@ -0,0 +1,96 @@
...
+    name: 'Second',
+    Component: TabContents,
+    options: {
+      ...DEFAULT_TAB_ROUTE_OPTIONS,
+      title: 'Second',
+    },
+  },
+];
+
+export function TabContents() {
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/index.tsx]: I think we can remove all exports here (except exporting TestStackSvmTabsSpecialEffects and default export of createScenario).

```suggestion
function TabContents() {
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/index.tsx [hunk 28]
```diff
@@ -0,0 +1,96 @@
...
+          contentInsetAdjustmentBehavior="automatic"
+          style={styles.fillParent}>
+          <ScrollViewContents />
+        </ScrollView>
+      </ScrollViewMarker>
+    </View>
+  );
+}
+
+export function TabsNavigation() {
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/index.tsx]: As mentioned above:
```suggestion
function TabsNavigation() {
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/index.tsx [hunk 29]
```diff
@@ -0,0 +1,96 @@
...
+  return <TabsContainer routeConfigs={TABS_ROUTE_CONFIGS} />;
+}
+
+export function ScrollViewContents(props: { elementCount?: number }) {
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/index.tsx]: As mentioned above:
```suggestion
function ScrollViewContents(props: { elementCount?: number }) {
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/index.tsx [hunk 30]
```diff
@@ -0,0 +1,96 @@
...
+      })}
+    </>
+  );
+}
+
+function HeuristicBreakingView() {
+  return <View collapsable={false} style={{ width: '100%', height: 0 }}></View>;
+}
+
+export default createScenario(App, scenarioDescription);
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/index.tsx]: Follows App name change has to by applied also here:

```suggestion
export default createScenario(TestSvmTabsScrollEdgeEffects, scenarioDescription);
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario-description.ts (hunk 4 above)
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario-description.ts]: Agree with above comments plus i think 'scroll' was omit: 

```suggestion
  name: 'SVM in Tabs - scroll edge effects',
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md [hunk 31]
```diff
@@ -0,0 +1,43 @@
...
+## Note
+
+Android part of the code is unimplemented yet, therefore it is expected not to work.
+
+iOS implementation doesn't currently support nested container interaction yet. 
+This test needs to be updated after such interaction is supported.
+
+## Steps
+
+1. Open the test.
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md]: I would make this a little bit more descriptive.

```suggestion
1. Launch the app and navigate to the **SVM in Stack & Tabs - tabs special effects** screen.
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md [hunk 32]
```diff
@@ -0,0 +1,44 @@
...
+
+Seemingly the edge effect is applied correctly no matter the integration with the scrollview marker. 
+Likely UITabBarController uses some different logic to UIViewController.contentScrollView to detect scrollview, 
+because even when I have had returned nil from the method, the edge effect had still been applied. 
+Nevertheless, I decided to include this test case here, just to make sure it works. 
+Also this allows us to test that different ScrollViewMarkers from different tab update the edge effect correctly on tab change.
+
+## Steps
+
+1. Open the test.
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md]:
```suggestion
1. Launch the app and navigate to the **SVM in Tabs - scroll edge effects** screen.
```

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md [hunk 33]
```diff
@@ -0,0 +1,44 @@
...
+
+- [ ] There should be two tabs: `Home` and `Second`.
+- [ ] `Home` tab should be selected.
+- [ ] `Hard` scroll-edge-effect should be applied (opqaue background of tab bar).
```
Comment (LKuchno) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md:35]: Maybe we can expand a bit on what is expected for `Hard`:

```suggestion
- [ ] `Hard` scroll-edge-effect should be applied (opaque background of tab bar with a hard cutoff and dividing line).
```

**Review by LKuchno — CHANGES_REQUESTED** [review]
I left a few comments with suggested changes to better align with our current test and scenario structures. Also changes corresponding to new export/import PR#4201 changes were propose. Some comments include open questions regarding the correct naming conventions.

File: ios/tabs/screen/RNSTabsScreenViewController.mm (hunk 7 above)
Comment (kkafar) [reply ios/tabs/screen/RNSTabsScreenViewController.mm:113]: No, I don't think there is. There might be once we add possibility to specify multiple scrollviews (different scrollviews for different edges), but it is not yet supported.

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario-description.ts (hunk 21 above)
Comment (kkafar) [reply apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario-description.ts:5]: no, that would be against naming convention IIRC. 
The name is alright IMO.

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md (hunk 31 above)
Comment (kkafar) [reply apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md]: I'm gonna apply this suggestion, however please consider that the more places we mention the test name in - the more possible places to go out of sync once the test name changes in the result of some refactor or so. I believe it's obvious from the context what test exactly is supposed to be opened, isn't it? On top of that, I don't require opening the test via the example app directly (I almost never do that). I render the test directly instead.

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md (hunk 25 above)
Comment (kkafar) [reply apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md]: I believe the indentation here to be correct. See the rendered version. 

What's more I believe the indentation of the `- [ ] <observation to verify>` lines to be incorrectly indented in almost all test cases. Shouldn't it be indicated visually that these points belong to the "action description points"? 

IIRC in markdown for the section to be considered "belonging to the outer section" you need double regular-indentation-size (usually two spaces -> 4 spaces).

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md (hunk 33 above)
Comment (kkafar) [reply apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md:35]: Dividing line? I haven't noticed such. Will take another look later 😄

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md (hunk 25 above)
Comment (LKuchno) [reply apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md]: Now I see, you have right. 
I will look at the indentation of the - [ ] <observation to verify> as I'm afraid that it can broke checkboxes in markdown.

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md (hunk 33 above)
Comment (LKuchno) [reply apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md:35]: It's actually taken from 'hard' scroll-edge-effect documentation I interpret it as visible cutoff tab bar  😅

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md [hunk 34]
```diff
@@ -0,0 +1,43 @@
...
+**OS test creation version:** 
+iOS 26.5
+
+## E2E test
+
+TBD
+
+## Prerequisites
+
+iOS: simulator with iOS 26+ is enough,
```
Comment (kmichalikk) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-svm-tabs-scroll-edge-effects/scenario.md]: nit: ,

File: apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md [hunk 35]
```diff
@@ -0,0 +1,43 @@
...
+This test needs to be updated after such interaction is supported.
+
+## Steps
+
+1. Launch the app and navigate to the **SVM in Stack & Tabs - tabs special effects** screen.
+
+- [ ] There should be two tabs: `Home` and `Stack`.
+- [ ] `Home` tab should be selected.
+
+2. Scrolldown a bit.
```
Comment (kmichalikk) [inline apps/src/tests/component-integration-tests/scroll-view-marker/test-stack-svm-tabs-special-effects/scenario.md]: nit: scroll down

**Review by kmichalikk — APPROVED** [review]

File: ios/gamma/scroll-view-marker/RNSScrollViewMarkerComponentView.mm (hunk 8 above)
Comment (kkafar) [reply ios/gamma/scroll-view-marker/RNSScrollViewMarkerComponentView.mm:162]: That won't do unfortunately. But I think I'll try to organise the registration in a bit different manner.

**Comment by kkafar** [comment]
@t0maboro (in context of: https://github.com/software-mansion/react-native-screens/pull/4191#issuecomment-4777012335)

I'm surprised it even started working. This PR does not cover integration with Stack. I'll do that in a follow-up.
